File: 백준/Silver/2108. 통계학/통계학.py
import sys
from collections import Counter

# 입력 읽기
n = int(sys.stdin.readline().rstrip())
number = []
sum = 0
for _ in range(n):
    num = int(sys.stdin.readline().rstrip())
    sum += num
    number.append(num)

# 리스트 정렬
number.sort()

# Counter를 이용한 빈도수 계산
cnt = Counter(number)

# 산술평균 계산 및 결과에 추가
result = [round(sum / n)]

# 중앙값 계산 및 결과에 추가
result.append(number[n // 2])

# 최빈값 계산
most_common_list = cnt.most_common()
most_common_list.sort(key=lambda x: (-x[1], x[0]))

if len(most_common_list) > 1 and most_common_list[0][1] == most_common_list[1][1]:
    result.append(most_common_list[1][0])
else:
    result.append(most_common_list[0][0])

# 범위 계산 (절대값 고려)
mi, ma = min(number), max(number)
result.append(ma - mi)

# 결과 출력
for elem in result:
    print(elem)

File: 백준/Silver/2108. 통계학/통계학.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.PriorityQueue;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st;
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		StringBuilder sb = new StringBuilder();

		int N = Integer.parseInt(br.readLine());

		int[] nums = new int[N];
		int[] count = new int[8001];
		
		double sum = 0;
		for (int i=0; i<N; i++) {
			int value = Integer.parseInt(br.readLine());
			nums[i] = value;
			sum += value;
			count[value+4000]++;
		}
		
		Arrays.sort(nums);
		
		// 1. 평균
		System.out.println(Math.round(sum/N));
		
		// 2. 중앙값
		if(N%2!=0) {
			System.out.println(nums[N/2]);
		} else {
			System.out.println(nums[N/2]);			
		}
		
		// 3. 최빈값
		int maxIdx = 0;
		int max = Integer.MIN_VALUE;
		for (int i=0; i<8001; i++) {
			if (count[i] > max) {
				max = count[i];
				maxIdx = i;
			}
		}
		
		int cnt = 0;
		for (int i=0; i<8001; i++) {
			if (count[i] == max && i != maxIdx) { //
				cnt++;
				System.out.println(i-4000);
				break;
			}
		
		}
		if(cnt==0) System.out.println(maxIdx-4000);
		
		
		// 4. 범위
		System.out.println(nums[N-1] - nums[0]);
		
		

		bw.write(sb.toString());
		bw.flush();
		bw.close();

	}
}
